Get rid of choosing one tablet by compaction

chaoyli opened the pull request.

morningman committed:
Fix bug that load error log is empty sometimes (#424)

_num_print_error_rows is not initialized

morningman committed:
Fix bug that load error log is empty sometimes (#424)

_num_print_error_rows is not initialized

Commit:
Get rid of choosing one tablet by compaction

yiguolei reviewed (dismissed):
LGTM

imay commented on be/src/olap/utils.cpp:
```diff
@@ -1104,11 +1104,11 @@ OLAPStatus Mutex::lock() {
-        VLOG(3) << "failed to got the mutex lock. err=" << strerror(errno);
+    int rv = pthread_mutex_trylock(&_lock);
+    if (rv != 0) {
+        VLOG(3) << "failed to got the mutex lock. error=" << strerror(rv);
```
use strerror_r to make it thread safe

chaoyli replied:
The return error is defined. So it is thread safe.

imay committed:
Add HttpClient class (#441)

Replace FileDownloader with HttpClient, this patch change clone_copy and
pusher's download.

imay committed:
Add HttpClient class (#441)

Replace FileDownloader with HttpClient, this patch change clone_copy and
pusher's download.

morningman committed:
Add log to detect empty load file (#445)

We find that a load file may not be generated for rollup tablet,
add a log to observe.

morningman committed:
Add log to detect empty load file (#445)

We find that a load file may not be generated for rollup tablet,
add a log to observe.

morningman committed:
Revert commit 'Add log to detect empty load file' (#447)

Looks like we still need to send push task without file
to Backend, or load job will fail.
Fix it later.

morningman committed:
Revert commit 'Add log to detect empty load file' (#447)

Looks like we still need to send push task without file
to Backend, or load job will fail.
Fix it later.

Commit:
Change PREFER_READER to PREFER_WRITING from _tablet_map_lock

imay committed:
Add UserFunctionCache to cache UDF's library (#453)

* Add UserFunctionCache to cache UDF's library

This patch replace LibCache with UserFunctionCache. LibCache use HDFS
URL to identify a UDF's Library, and when BE process restart all of
downloaded library should be loaded another time. We use function id
corresponding to a library, and when process restart, all downloaded
libraries can be loaded without another downloading.

* update

Commit:
Change license of murmur_hash
